src/models/state/mod.rs

```diff
@@ -1472,6 +1482,92 @@ impl GlobalState {
         // Ok(())
     }
 
+    /// Returns (parent, child) pair of blocks.
+    pub(crate) async fn fetch_block_pair(&self, child_digest: Digest) -> Option<(Block, Block)> {
```

Comment:
I think this function crashes if `child_digest` is the genesis block. We check that it's not in the function `response_to_sync_challenge`, but if this function is public, it should probably return `None` in that case.